modules/solace/solace.go

```diff
@@ -0,0 +1,291 @@
+package solace
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/go-connections/nat"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+type SolaceContainer struct {
+	Container      testcontainers.Container
+	Username       string
+	Password       string
+	Vpn            string
+	queues         map[string][]string // queueName -> topics
+	WithClientCert bool
+	Image          string
+	ctx            context.Context
+	exposedPorts   []string
+	envVars        map[string]string
+	shmSize        int64
+}
+
+// WithCredentials sets the client credentials (username, password)
+func (s *SolaceContainer) WithCredentials(username, password string) *SolaceContainer {
+	s.Username = username
+	s.Password = password
+	return s
+}
+
+// WithVpn sets the VPN name
+func (s *SolaceContainer) WithVpn(vpn string) *SolaceContainer {
+	s.Vpn = vpn
+	return s
+}
+
+// WithQueue subscribes a given topic to a queue (for SMF/AMQP testing)
+func (s *SolaceContainer) WithQueue(queueName, topic string) *SolaceContainer {
+	// Store queue/topic mapping in a new field for later use (e.g., CLI script generation)
+	if s.queues == nil {
+		s.queues = make(map[string][]string)
+	}
+	s.queues[queueName] = append(s.queues[queueName], topic)
+	return s
+}
+
+// WithEnv allows adding or overriding environment variables
+func (s *SolaceContainer) WithEnv(env map[string]string) *SolaceContainer {
+	if s.envVars == nil {
+		s.envVars = make(map[string]string)
+	}
+	for k, v := range env {
+		s.envVars[k] = v
+	}
+	return s
+}
+
+// WithExposedPorts allows adding extra exposed ports
+func (s *SolaceContainer) WithExposedPorts(ports ...string) *SolaceContainer {
+	s.exposedPorts = append(s.exposedPorts, ports...)
+	return s
+}
+
+const (
+	DefaultImage = "solace/solace-pubsub-standard:latest"
+	DefaultVpn   = "default"
+	DefaultUser  = "root"
+	DefaultPass  = "password"
+)
+
+func NewSolaceContainer(ctx context.Context, image string) *SolaceContainer {
```

Comment:
issue: as per my comment above, this constructor function is not needed anymore.

Reply:
burned with fire 🔥